### DISCUSSION TITLE
Proposal: remove `created` event

### DISCUSSION THREAD

**Discussion by rigor789** [NativeScript/rfcs#48] [1. Ideas & Discussions]
Core currently emits a `created` event in the ViewBase constructor, however this is not used within core, and using it in user code is not all that useful either because it's emitted before you get a chance to add a listener, so essentially you can't directly listen to the `created` event on a view. The only way to listen to it is through global listeners, ie.:

```ts
Observable.on('created', (args) => {
  console.log('created', args.object)
})
```

Since this fires in the constructor, you don't have access to the view `id`, only the `_domId`.

https://github.com/NativeScript/NativeScript/blob/8d25d251cd1c311d3f73e526bf2fd4ee0b2be0b8/packages/core/ui/core/view-base/index.ts#L472

Proposal: remove this event from core, it seems useless and only adds a slight overhead to all view creation.

If you know a valid use-case for why we should keep this, please comment here!

/cc @NativeScript/tsc

**Comment by shirakaba** [comment]
I say remove it. When reimplementing events from the ground up in `nativescript-html`, I found I wasn't able to support listening for events in the constructor, so I just ignored the `created` events, and things still worked fine. Also found no trace of it elsewhere in the monorepo.
